data-migration/migrate-with-more-columns-downstream.md

```diff
@@ -0,0 +1,108 @@
+- [Import and Merge Large MySQL Sharding Schemas and Sharding Tables to TiDB (Greater Than 1 TiB)](/data-migration/migrate-shared-mysql-tidb-above-tb.md)
+
+## Using DM to migrate data to a downstream TiDB table with more columns
+
+When replicating the upstream binlog, DM tries to use the current table schema of the downstream to parse the binlog and generate the corresponding DML statements. If the column number of the table in the upstream binlog does not match the column number in the downstream table schema, the following error occurs:
+
+```json
+"errors": [
+    {
+        "ErrCode": 36027,
+        "ErrClass": "sync-unit",
+        "ErrScope": "internal",
+        "ErrLevel": "high",
+        "Message": "startLocation: [position: (mysql-bin.000001, 2022), gtid-set:09bec856-ba95-11ea-850a-58f2b4af5188:1-9 ], endLocation: [ position: (mysql-bin.000001, 2022), gtid-set: 09bec856-ba95-11ea-850a-58f2b4af5188:1-9]: gen insert sqls failed, schema: log, table: messages: Column count doesn't match value count: 3 (columns) vs 2 (values)",
+        "RawCause": "",
+        "Workaround": ""
+    }
+]
+```
+
+The following is an example upstream table schema:
+
+```sql
+# Upstream table schema
+CREATE TABLE `messages` (
+  `id` int(11) NOT NULL,
+  PRIMARY KEY (`id`)
+)
+```
+
+The following is an example downstream table schema:
+
+```sql
+# Downstream table schema
+CREATE TABLE `messages` (
+  `id` int(11) NOT NULL,
+  `message` varchar(255) DEFAULT NULL, # This is the additional column that only exists in the downstream table.
+  PRIMARY KEY (`id`)
+)
+```
+
+When DM tries to use the downstream table schema to parse the binlog event generated by the upstream, DM reports the above `Column count doesn't match` error.
+
+In such cases, you can use the `operate-schema` command to set a table schema for the table to be migrated from the data source. The specified table schema needs to correspond to the binlog event data to be replicated by DM. If you are migrating sharded tables, for each sharded table, you need to set a table schema in DM to parse binlog event data as follows:
+
+1. Create a `.sql` file in DM and add the `CREATE TABLE` statement that corresponds to the upstream table schema to the file. For example, save the following table schema to `log.messages.sql`.
+
+    ```sql
+    # Upstream table schema
+    CREATE TABLE `messages` (
+    `id` int(11) NOT NULL,
+    PRIMARY KEY (`id`)
+    )
+    ```
+
+2. Use the `operate-schema` command to set the table schema for the table to be migrated from the data source (At this time, the data migration task should be in the Paused state due to the above `Column count doesn't match` error).
+
+    {{< copyable "shell-regular" >}}
+
+    ```
+    tiup dmctl --master-addr ${advertise-addr} operate-schema set -s ${mysql-id} ${task-name} -d ${database-name} -t ${table-name} ${schema-file}
```

Comment:
Prefer `source-id` rather than `mysql-id`. Please update the Chinese doc aslo.

Reply:
Done in https://github.com/pingcap/docs/pull/7248/commits/7793f8d30562c2d26320ffcbb584b30aa0ab9058.